packages/docs/docs/docs/architecture.md

```diff
@@ -0,0 +1,97 @@
+
+A Package contract tracks all the versions of a given EVM package. Following the example above, one package could be the "application package" associated to the name "my-application" containing all the contracts for version "0.0.1" of your application, and all the contracts for version "0.0.2" as well. Alternatively, another package could be an "EVM package" associated to the name "openzeppelin-zos" which contains a large number of versions "x.y.z" each of which contains a given set of contracts.
+
+The versions are mapped by a semver hash to `Version` structs:
+
+#### Package.sol
+```solidity
+
+  // .. 
+
+  mapping (bytes32 => Version) internal versions;
+
+  // .. 
+
+  struct Version {
+    uint64[3] semanticVersion;
+    address contractAddress;
+    bytes contentURI; 
+  }
+
+  // ..
+```
+
+## [ImplementationDirectory](https://github.com/zeppelinos/zos/blob/master/packages/lib/contracts/application/ImplementationDirectory.sol)
+
+A version's `contractAddress` is an instance of the `ImplementationDirectory` contract, which is basically a mapping of contract aliases (or names) to deployed implementation instances. Continuing the example, your project's "my-application" package for version "0.0.1" could contain a directory with the following contracts:
+
+#### Directory for version "0.0.1" of the "my-application" package
+* Alias: "MainContract", Implementation: "0x0B06339ad63A875D4874dB7B7C921012BbFfe943" 
+* Alias: "MyToken", Implementation: "0x1b9a62585255981c85Acec022cDaC701132884f7"
+
+While version "0.0.2" of the "my-application" package could look like this:
+
+#### Directory for version "0.0.2" of the "my-application" package 
+* Alias: "MainContract", Implementation: "0x0B06339ad63A875D4874dB7B7C921012BbFfe943" 
+* Alias: "MyToken", Implementation: "0x724a43099d375e36c07be60c967b8bbbec985dc8" <--- this changed
+
+Notice how version "0.0.2" uses a new implementation for the "MyToken" contract.
+
+Likewise, different versions of the "openzeppelin-zos" EVM package could contain different implementations for persisting aliases such as "ERC20", "ERC721", etc.
+
+An `ImplementationDirectory` is a contract that adopts the `ImplemetationProvider` interface, which simply requires that for a given contract alias or name, the deployed address of a contract is provided. In this particular implementation of the interface, an `ImplementationDirectory` can be frozen, indicating that it will no longer be able to set or unset additional contracts and aliases.
+This is particularly helpful for making oficial releases of EVM packages, where the immutability of the package is guaranteed.
+
+Other implementations of the interface could provide contracts without such a limitation, which makes the architecture pretty flexible, yet secure.
+
+## Overview
+
+The following diagram illustrates the interface of the application contracts:
+
+![zOS 2.x UML](img/zos2.png)
```

Comment:
It's not an architecture document by @ajsantander if it doesn't have an UML diagram <3